OVERVIEW.md

```diff
@@ -0,0 +1,66 @@
+# Overview #
+
+Here it is described a brief overview of xdist's internal architecture.
+
+
+`xdist` works by spawning one or more **worker nodes**, which are controlled
+by the **master node**. Each **worker node** is responsible for performing 
+a full test collection and afterwards running tests as dictated by the **master node**.
+
+The execution flow is:
+
+1. **master node** spawns one or more **worker nodes** at the begginning of
+   the test session. The communication between **master** and **worker** nodes makes use of 
+   [execnet](http://codespeak.net/execnet/) and its [gateways](http://codespeak.net/execnet/basics.html#gateways-bootstrapping-python-interpreters).
+   The actual interpreters executing the code for the **worker nodes** might
+   be remote or local. 
+  
+1. Each **worker node** itself is a mini pytest runner. **workers** at this
+   point perform a full test collection, sending back the collected 
+   test-ids back to the **master node** which does not
+   perform any collection itself.
```

Comment:
Architecture question: Why does each worker do its own collection, as opposed to having the master collect once and distribute from that collection to the workers?

This would drop the need for the master to do the sanity check you describe later, and would perhaps also facilitate the work we are discussing in #18.